frontend/src/components/widgets/FileUploader/FileUploader.tsx

```diff
@@ -129,135 +177,131 @@ class FileUploader extends React.PureComponent<Props, State> {
+  ): void => {
+    const fileId = event ? event.currentTarget.id : id
+    const file = this.state.files.find(file => file.id === fileId)
+    if (fileId && file) {
+      file.status = FileStatuses.DELETING
+      if (file.cancelToken) {
+        file.cancelToken.cancel()
+      }
+      this.setState({ files: [...this.state.files] })
 
-    // Hack to hide drag-and-drop message and leave space for filename.
-    let overrides: any = fileUploaderOverrides
-    let filenameText = ""
+      // File was not uploaded. No need to make a HTTP call.
+      if (file.errorMessage || file.cancelToken) {
+        this.removeFile(fileId)
+        return
+      }
+    } else {
+      const errorMessage = "File not found. Please try again."
+      this.setState({
+        status: FileStatuses.ERROR,
+        errorMessage,
+      })
 
-    if (status === "UPLOADED") {
-      overrides = { ...overrides }
-      overrides.ContentMessage = { ...overrides.ContentMessage }
-      overrides.ContentMessage.style = { ...overrides.ContentMessage.style }
-      overrides.ContentMessage.style.visibility = "hidden"
-      overrides.ContentMessage.style.overflow = "hidden"
-      overrides.ContentMessage.style.height = "0.625rem" // half of lineHeightTight
+      return
+    }
 
-      overrides.ContentSeparator = { ...overrides.ContentSeparator }
-      overrides.ContentSeparator.style.visibility = "hidden"
+    this.props.uploadClient
+      .delete(this.props.element.get("id"), fileId)
+      .then(() => this.removeFile(fileId))
+  }
 
-      if (multipleFiles) {
-        filenameText = this.state.acceptedFiles
-          .map(file => file.name)
-          .join(", ")
-      } else {
-        filenameText = this.state.acceptedFiles[0].name
-      }
-    }
+  private removeFile = (fileId: string): void => {
+    const filteredFiles = this.state.files.filter(file => file.id !== fileId)
+    this.setState({
+      status: filteredFiles.length
+        ? FileStatuses.UPLOADED
+        : FileStatuses.READY,
+      errorMessage: undefined,
+      files: filteredFiles,
+    })
+  }
 
-    return (
-      <>
-        {status === "UPLOADED" && (
-          <div className="uploadOverlay uploadDone">
-            <span className="body">{filenameText}</span>
-          </div>
-        )}
-        <FileUploaderBaseui
-          onDrop={this.dropHandler}
-          errorMessage={errorMessage}
-          accept={accept.length === 0 ? undefined : accept}
-          disabled={this.props.disabled}
-          overrides={overrides}
-          multiple={multipleFiles}
-        />
-      </>
+  private onUploadProgress = (
+    progressEvent: ProgressEvent,
+    file: ExtendedFile
+  ): void => {
+    file.progress = Math.round(
+      (progressEvent.loaded * 100) / progressEvent.total
     )
+
+    this.setState(state => {
+      const files = state.files.map(uploadingFile =>
+        uploadingFile.id === file.id ? file : uploadingFile
+      )
+
+      return { files }
+    })
```

Comment:
updater function `setState` takes a function rather than an object.

```
this.setState({
  files: this.state.files.map(uploadingFile =>
        uploadingFile.id === file.id ? file : uploadingFile
      )
})
```